app/src/main/java/com/duckduckgo/app/fire/GetCookieHostsToPreserve.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2020 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.fire
+
+import com.duckduckgo.app.fire.fireproofwebsite.data.FireproofWebsiteDao
+
+class GetCookieHostsToPreserve(private val fireproofWebsiteDao: FireproofWebsiteDao) {
```

Comment:
this returns a list of hosts which might include `.com`. I don't fully understand the logic here, so just raising the query that this is expected and OK?

Reply:
yes, it's expected and I can add a comment there. The reason is that we can't differentiate between domains and top-level domains (e.g: .com, .co.uk, .io, ...), but no cookies can exist for top-level domain (RFC2109).
The cookie host needs to be a FQHN (fully-qualified host name: host + top-level domain)